### PR TITLE
Minor refactors

### PR DESCRIPTION
**Pull request by paplorinc**
I noticed some inspections could be applied to the codebase, so I wend ahead and did just that (in separate  commits).

### CODE REVIEW

File: vavr/src/main/java/io/vavr/collection/AbstractMultimap.java [hunk 1]
```diff
@@ -579,22 +579,26 @@ public String toString() {
...
-            throw new IllegalStateException("Unknown ContainerType: " + containerType);
-        }
+        final Supplier<Collection<V>> javaContainerSupplier = getJavaContainerSupplier();
         for (Tuple2<K, V> t : this) {
             javaMap.computeIfAbsent(t._1, k -> javaContainerSupplier.get()).add(t._2);
         }
         return javaMap;
     }
 
+    private Supplier<Collection<V>> getJavaContainerSupplier() {
```
Comment (chb0github) [inline vavr/src/main/java/io/vavr/collection/AbstractMultimap.java]: Better would be to have this as a parameter in the enum constructor and add a getter - then the 

```
default:
                throw new IllegalStateException("Unknown ContainerType: " + containerType);
```

Can't possibly occur. 

```
enum ContainerType {

        SET(
                (Traversable<?> set, Object elem) -> ((Set<Object>) set).add(elem),
                (Traversable<?> set, Object elem) -> ((Set<Object>) set).remove(elem),
                HashSet::new,
        )

        ContainerType(BiFunction<Traversable<?>, Object, Traversable<?>> add,
                BiFunction<Traversable<?>, Object, Traversable<?>> remove,
                 Supplier<Collection<?>> instantiator) {
            this.add = add;
            this.remove = remove;
            this.instantiator = instantiator;
        }

        public Supplier<Collection<?>> getInstantiator() {
             return instantiator;
        }

Comment (paplorinc) [reply vavr/src/main/java/io/vavr/collection/AbstractMultimap.java]: Interesting, lemme' check it, thank you! :)

Comment (paplorinc) [reply vavr/src/main/java/io/vavr/collection/AbstractMultimap.java]: Cool idea @chb0github, applied it in https://github.com/vavr-io/vavr/pull/2112/commits/b5205538133f0ec732e2161bd5b5e5b38c35fa2b

Comment (danieldietrich) [reply vavr/src/main/java/io/vavr/collection/AbstractMultimap.java]: @chb0github nice one!

**Pull request merged**